util/ssl_support.c

```diff
@@ -380,7 +380,7 @@ int SBUF2_FUNC(ssl_new_ctx)(SSL_CTX **pctx, ssl_mode mode, const char *dir,
 #endif /* HAVE_CRL */
 
 #ifndef OPENSSL_NO_ECDH
-#if OPENSSL_VERSION_NUMBER >= 0x10200000L
+#if OPENSSL_VERSION_NUMBER >= 0x10002000L
```

Comment:
This should actually be:
```suggestion
#if OPENSSL_VERSION_NUMBER < 0x10100000L
```

The [changelog](https://github.com/openssl/openssl/blob/master/CHANGES.md#changes-between-102h-and-110--25-aug-2016) for version 1.1 has this to say about this function:
```
SSL_{CTX_}set_ecdh_auto() has been removed and ECDH is support is always enabled now. If you want to disable the support you should exclude it using the list of supported ciphers. This also means that the "-no_ecdhe" option has been removed from s_server.
```

Reply:
Yes. Your approach is definitely more complete. `SSL_CTX_set_ecdh_auto()` isn't gone in 1.1.0. It's made a no-op ([man page](https://www.openssl.org/docs/manmaster/man3/SSL_CTX_set_ecdh_auto.html)). So what Carlo has is probably good enough.